### PR TITLE
Remove Module#attr warning with given boolean argument [Feature #17099]

### PR DESCRIPTION
**Pull request by S-H-GAMELINKS**
Remove `Module#attr` warning(with given boolean argument)

ref: [Feature #17099 Remove Module#attr warning with given boolean argument](https://bugs.ruby-lang.org/issues/17099)

### CODE REVIEW

File: spec/ruby/core/module/attr_spec.rb [hunk 1]
```diff
@@ -33,66 +33,6 @@ def initialize
     o.send(:attr3).should == "test3"
   end
 
-  it "creates a setter for the given attribute name if writable is true" do
```
Comment (eregon) [inline spec/ruby/core/module/attr_spec.rb:36]: Specs should not be removed, instead they should be under `ruby_version_is ""..."3.1" do` guard.

Comment (S-H-GAMELINKS) [reply spec/ruby/core/module/attr_spec.rb:36]: OK, I'll fix it.

Comment (eregon) [reply spec/ruby/core/module/attr_spec.rb:36]: Looks good now, but there should be a new `it` spec with `ruby_version_is "3.1" do` describing the new behavior, which likely is just some kind of error.

**Comment by S-H-GAMELINKS** [comment]
Rebased this pull request to head of master(0af9eab08e41167241150439e82041504e2ad26c)

File: spec/ruby/core/module/attr_spec.rb [hunk 2]
```diff
@@ -136,14 +138,16 @@ def initialize
...
+      -> {
+        Class.new { attr :foo, true }
+      }.should complain(/boolean argument is obsoleted/, verbose: true)
+    end
 
-  it "is a public method" do
-    Module.should have_public_instance_method(:attr, false)
+    it "is a public method" do
+      Module.should have_public_instance_method(:attr, false)
+    end
```
Comment (eregon) [inline spec/ruby/core/module/attr_spec.rb:150]: This should not be inside the version guard

Comment (S-H-GAMELINKS) [reply spec/ruby/core/module/attr_spec.rb]: @eregon 
Thanks for your comments!
I've done it(https://github.com/ruby/ruby/pull/3386/commits/87dba5ef95519b9302b934b958a87db2751ecc9c)

**Comment by eregon** [comment]
I think we need an explicit approval on https://bugs.ruby-lang.org/issues/17099, you can add it to the dev meeting, because it is a breaking change (which was warned before, but deprecation warnings are not even shown by default nowadays)

**Comment by S-H-GAMELINKS** [comment]
@eregon 
Sorry for late reply.

Already disscussed this change in developer meeting(one years ago).
But, if you think need to more disscussing to this ticket, I add this ticket to  developer meeting.

FYI
https://github.com/ruby/dev-meeting-log/blob/master/DevelopersMeeting20210113Japan.md#feature-17099-remove-boolean-argument-and-warning-from-moduleattr-s_h_

**Comment by eregon** [comment]
From that link:
> [Feature #17099] Remove boolean argument and warning from Module#attr (S_H_)
>    It has been warned since 1.9.1.
> Conclusion:
>     naruse: Don't remove in 3.1. Maybe remove in 3.2 (I don't have opinion in 3.2)

So it's still not a "yes/go ahead" (which is normally also written on the ticket), so yeah please add it to the dev meeting.

**Pull request closed without merging**